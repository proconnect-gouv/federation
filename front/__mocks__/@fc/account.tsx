import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div data-mockid={'AccountProvider'}>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
