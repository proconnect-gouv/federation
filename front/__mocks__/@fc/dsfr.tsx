export const FranceConnectButton = jest.fn(() => <div>FranceConnectButton</div>);

export const SimpleButton = jest.fn(() => <div>SimpleButton</div>);

export const CheckboxInput = jest.fn(() => <div>CheckboxInput</div>);

export const ToggleInput = jest.fn(() => <div>ToggleInput</div>);

export const ApplicationLayout = jest.fn(() => <div>ApplicationLayout</div>);

export const LinkComponent = jest.fn(() => <div>LinkComponent</div>);

export const SearchBarComponent = jest.fn(() => <div>SearchBarComponent</div>);

export const BadgeComponent = jest.fn(() => <div>BadgeComponent</div>);

export const AlertComponent = jest.fn(({ children }) => (
  <div data-mockid={'AlertComponent'}>
    <div>AlertComponent</div>
    <div>{children}</div>
  </div>
));

export const PaginationComponent = jest.fn(() => <div>PaginationComponent</div>);

export const AccordionComponent = jest.fn(({ children }) => (
  <div data-mockid={'AccordionComponent'}>
    <h3>AccordionComponent</h3>
    <div>{children}</div>
  </div>
));

export const AccordionGroupComponent = jest.fn(({ children }) => (
  <div data-mockid={'AccordionGroupComponent'}>
    <h3>AccordionGroupComponent</h3>
    <div>{children}</div>
  </div>
));

export const Breakpoints = {
  LG: 1248,
  MD: 992,
  SM: 768,
  XS: 576,
};

export const Sizes = {
  LARGE: 'lg',
  MEDIUM: 'md',
  SMALL: 'sm',
};

export const AlertTypes = {
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
};

export const IconPlacement = {
  LEFT: 'left',
  RIGHT: 'right',
};

export const Priorities = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
};

export const ButtonTypes = {
  BUTTON: 'button',
  RESET: 'reset',
  SUBMIT: 'submit',
};
