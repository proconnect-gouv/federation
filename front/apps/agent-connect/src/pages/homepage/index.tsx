/* istanbul ignore file */

/**
 * Tested with cypress snapshots
 */
import React from 'react';
import { useSelector } from 'react-redux';

import FCAApiContext from '../../components/fca-api-context';
import { RootState } from '../../types';
import IdentityProvidersUserHistory from './idp-user-history';
import Search from './search';
import ServiceProviderName from './sp-name';

const HomePage = React.memo((): JSX.Element => {
  const identityProvidersHistory = useSelector(
    (state: RootState) => state.identityProvidersHistory,
  );

  const showUserHistory = identityProvidersHistory?.length > 0;

  return (
    <FCAApiContext>
      <React.Fragment>
        <ServiceProviderName />
        {showUserHistory && (
          <IdentityProvidersUserHistory items={identityProvidersHistory} />
        )}
        <Search />
      </React.Fragment>
    </FCAApiContext>
  );
});

HomePage.displayName = 'HomePage';

export default HomePage;
