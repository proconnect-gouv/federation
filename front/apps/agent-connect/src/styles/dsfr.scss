// @TODO a supprimer une fois la lib dsfr partag√©e entre les apps front

.mr8 {
  margin-right: 8px;
}

.is-block {
  display: block;
}

.flex-columns {
  display: flex;
  flex-direction: row;
}

.items-center {
  align-items: center;
}

a.no-underline {
  text-decoration: none;
}

.bg-blue-france-100 {
  background-color: #f2f2f9;
}

.is-blue-agentconnect {
  color: #0579ee;
}

.fr-display-xs {
  font-size: 2.5rem;
  line-height: 3rem;
}

h1 {
  font-size: 2rem;
  line-height: 2.5rem;
}

.content-wrapper-sm {
  margin-left: auto;
  margin-right: auto;
}

.my32 {
  margin-bottom: 2rem;
  margin-top: 2rem;
}

.mb32 {
  margin-bottom: 2rem;
}

.px120 {
  padding-left: 7.5rem;
  padding-right: 7.5rem;
}

.fr-text-lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.p24 {
  padding: 1.5rem;
}

.no-no-padding {
  padding: 0 !important;
}

@media (min-width: 768px) {
  .fr-display-xs {
    font-size: 3rem;
    line-height: 3.5rem;
  }

  h1 {
    font-size: 2.5rem;
    line-height: 3rem;
  }

  .content-wrapper-sm {
    max-width: 792px;
  }

  .fr-text-lg {
    font-size: 1.25rem;
    line-height: 2rem;
  }
}
