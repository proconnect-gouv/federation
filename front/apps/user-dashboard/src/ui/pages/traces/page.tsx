import Introduction from './introduction';
import TracesList from './traces-list';
import UserWelcome from './user-welcome';

const TracesPageComponent = () => (
  <div className="content-wrapper-lg px16" id="page-container">
    <UserWelcome />
    <Introduction />
    <TracesList />
  </div>
);

TracesPageComponent.displayName = 'TracesPageComponent';

export default TracesPageComponent;
