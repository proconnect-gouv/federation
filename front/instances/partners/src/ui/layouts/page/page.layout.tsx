import React from 'react';
import { Outlet } from 'react-router-dom';

export const PageLayout = React.memo(() => (
  <main className="fr-container fr-py-8v">
    <div className="fr-grid-row fr-grid-row--center">
      <Outlet />
    </div>
  </main>
));

PageLayout.displayName = 'PageLayout';
