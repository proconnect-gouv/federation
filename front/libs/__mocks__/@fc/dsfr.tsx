export const FranceConnectButton = jest.fn(() => <div>FranceConnectButton</div>);

export const SimpleButton = jest.fn(() => <div>SimpleButton</div>);

export const CheckboxInput = jest.fn(() => <div>CheckboxInput</div>);

export const ToggleInput = jest.fn(() => <div>ToggleInput</div>);

export const ApplicationLayout = jest.fn(() => <div>ApplicationLayout</div>);

export const LinkComponent = jest.fn(() => <div>LinkComponent</div>);

export const SearchBarComponent = jest.fn(() => <div>SearchBarComponent</div>);

export const Sizes = {
  LARGE: 'lg',
  MEDIUM: 'md',
  SMALL: 'sm',
};

export const AlertTypes = {
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
};

export const AlertComponent = jest.fn(() => <div>Alert</div>);

export const PaginationComponent = jest.fn(() => <div>PaginationComponent</div>);

export { ButtonTypes, IconPlacement, Priorities } from '@fc/dsfr';
