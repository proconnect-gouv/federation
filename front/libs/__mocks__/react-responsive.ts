export const useMediaQuery = jest.fn(() => true);
