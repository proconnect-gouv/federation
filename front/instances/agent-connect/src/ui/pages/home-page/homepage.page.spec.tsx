import { render } from '@testing-library/react';
import { useMediaQuery } from 'react-responsive';

import { LinkComponent, Sizes } from '@fc/dsfr';

import {
  SearchComponent,
  ServiceProviderNameComponent,
  UserHistoryComponent,
} from '../../components';
import { HomePage } from './homepage.page';

jest.mock('@fc/dsfr');
jest.mock('../../components/search/search.component');
jest.mock('../../components/user-history/user-history.component');
jest.mock('../../components/service-provider-name/service-provider-name.component');

describe('HomePage', () => {
  it('should math the snapshot, in a desktop viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(<HomePage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should math the snapshot, in a mobile viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(false);

    // when
    const { container } = render(<HomePage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should call a LinkComponent with params', () => {
    // when
    render(<HomePage />);

    // then
    expect(LinkComponent).toHaveBeenCalledTimes(1);
    expect(LinkComponent).toHaveBeenCalledWith(
      expect.objectContaining({
        href: 'https://agentconnect.gouv.fr/aide',
        icon: 'question-fill',
        label: 'J’ai besoin d’aide',
        size: Sizes.LARGE,
      }),
      {},
    );
  });

  it('should call ServiceProviderNameComponent', () => {
    // when
    render(<HomePage />);

    // then
    expect(ServiceProviderNameComponent).toHaveBeenCalled();
  });

  it('should call UserHistoryComponent', () => {
    // when
    render(<HomePage />);

    // then
    expect(UserHistoryComponent).toHaveBeenCalled();
  });

  it('should call SearchComponent', () => {
    // when
    render(<HomePage />);

    // then
    expect(SearchComponent).toHaveBeenCalled();
  });
});
