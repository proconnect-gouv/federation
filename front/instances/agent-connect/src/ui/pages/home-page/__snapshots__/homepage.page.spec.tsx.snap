// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should math the snapshot, in a desktop viewport 1`] = `
<div>
  <div>
    <div>
      React Helmet
    </div>
  </div>
  <div
    class="fr-mt-8w"
  >
    <div>
      ServiceProviderNameComponent
    </div>
    <div>
      UserHistoryComponent
    </div>
    <div>
      SearchComponent
    </div>
    <div
      class="text-center fr-mt-6w"
    >
      <div>
        LinkComponent
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage should math the snapshot, in a mobile viewport 1`] = `
<div>
  <div>
    <div>
      React Helmet
    </div>
  </div>
  <div
    class="fr-mt-4w"
  >
    <div>
      ServiceProviderNameComponent
    </div>
    <div>
      UserHistoryComponent
    </div>
    <div>
      SearchComponent
    </div>
    <div
      class="text-center fr-mt-6w"
    >
      <div>
        LinkComponent
      </div>
    </div>
  </div>
</div>
`;
